CONTRIBUTING.md

```diff
@@ -0,0 +1,151 @@
+Contributing to Mypy
+====================
+
+Welcome!  Mypy is a community project that aims to work for a wide
+swath of Python users and Python codebases.  If you're trying Mypy on
+your Python code, your experience and what you can contribute is
+important to the project's success.
+
+
+Getting started, building, and testing
+--------------------------------------
+
+If you haven't already, take a look at the project's README file and
+the [Mypy documentation](http://mypy.readthedocs.org/en/latest/), and
+try adding type annotations to your file and type-checking it with Mypy.
+
+
+Discussion
+----------
+
+If you've run into behavior in Mypy you don't understand, or you're
+having trouble working out a good way to apply it to your code, or
+you've found a bug or would like a feature it doesn't have, we want to
+hear from you!
+
+Our main forum for discussion is the project's [GitHub issue
+tracker](https://github.com/python/mypy/issues).  This is the right
+place to start a discussion of any of the above or most any other
+topic concerning the project.
+
+We have an IRC channel, `#python-mypy` on irc.freenode.net.  This is
+lightly used, but some Mypy core developers are almost always present;
+feel free to find us there and we're happy to chat.  Substantive
+technical discussion will be directed to the issue tracker.
+
+#### Code of Conduct
+
+Everyone participating in the Mypy community, and in particular in our
+issue tracker, pull requests, and IRC channel, is expected to treat
+other people with respect and more generally to follow the guidelines
+articulated in the [Python Community Code of
+Conduct](https://www.python.org/psf/codeofconduct/).
+
+
+Submitting Changes
+------------------
+
+Even more excellent than a good bug report is a fix for a bug, or the
+implementation of a much-needed new feature.  We'd love to have your
+contributions.
+
+We use the usual GitHub pull-request flow, which may be familiar to
```

Comment:
I'd write "comment on a pull request", but "your pull-request comment" -- hyphenating the phrase when it's used as a modifier but not when it stands alone in the role of a noun.

This is an area of writing style where different references give slightly different recommendations -- some recommend applying this to more compounds, some to fewer -- though all agree that there are many phrases where this is the preferred style. Here's [Wikipedia's description](https://en.wikipedia.org/wiki/Compound_modifier#Compound_adjectives), and here's a [somewhat polemical treatment](http://sites.utexas.edu/legalwriting/2014/07/21/the-compound-modifier-hyphen-connects-and-clarifies/) by a law-school writing professor.

I generally apply the same approach as that legal-writing professor does, hyphenating nearly all compound modifiers outside some well-defined classes. He cites Garner, a contemporary usage authority I've seen recommended before, as recommending the same approach.